kubectl-plugin/pkg/cmd/log/log.go

```diff
@@ -123,12 +125,13 @@ func (options *ClusterLogOptions) Validate() error {
 	}
 
 	if options.outputDir == "" {
-		fmt.Fprintln(options.ioStreams.Out, "No output directory specified, creating dir under current directory using cluster name.")
+		fmt.Fprintln(options.ioStreams.Out, "No output directory specified, creating dir under current directory using resource name.")
 		options.outputDir = options.ResourceName
 		err := os.MkdirAll(options.outputDir, 0o755)
 		if err != nil {
 			return fmt.Errorf("could not create directory with cluster name %s: %w", options.outputDir, err)
 		}
+		genereatedDirectory = true
```

Comment:
Instead of tracking a global env var here, can you check `options.outputDir == ""` in Run()?

Reply:
The `options.outputDir` is filled in Validations if the user does not input a outputDir. It then becomes difficult to know if we were the one that generated the dir or its user input.